### ISSUE TITLE
pkg_resources.extern.packaging.markers.UndefinedEnvironmentName Exception

### ISSUE THREAD

**Issue by willypuzzle** [python-babel/babel#501]
When I try the command:

$pybabel extract -F babel.cfg -o messages.pot .

I receive this output:

> ...
> extracting messages from app/templates/auth/login.html (extensions="jinja2.ext.autoescape,jinja2.ext.with_")
> Traceback (most recent call last):
>   File "/var/www/python/flask-foundation/env/bin/pybabel", line 11, in <module>
>     sys.exit(main())
> ...
> pkg_resources.extern.packaging.markers.UndefinedEnvironmentName: 'extra' does not exist in evaluation environment.

Why? I read it can depend from pip but I have the last upgraded version. This is my babel.cfg file:

```
[python: **.py]
[jinja2: **/templates/**.html]
extensions=jinja2.ext.autoescape,jinja2.ext.with_
```

I use python 3.

**Comment by halfnibble** [comment]
I'm having this issue now. Latest Ubuntu LTS and it's latest Python3 and pip.

**Comment by benselme** [comment]
Could it be the same problem as in this issue ?
https://github.com/getsentry/sentry/issues/3143

**Comment by Mark-Wellings** [comment]
I'm afraid it's not the same issue. Just tried downgrading and it didnt work. If you dont reference jinja2 it does run, but for me creates an empty messages.pot
`[python: **/FlaskApp/**.py]

encoding = utf-8
extensions=jinja2.ext.autoescape,jinja2.ext.with_,webassets.ext.jinja2.AssetsExtension
silent=False`

**Comment by akx** [comment]
@Mark-Wellings Could you create a repo with a minimal test case that exhibits this problem?

**Comment by Mark-Wellings** [comment]
I think my issue was related to how my app is organized. Reading all *.py files on the server gave me an error. Now I've got it set up so that it only reads __init__.py:
`[python: **__.py]
[jinja2: **.html]
encoding = utf-8
extensions=jinja2.ext.autoescape,jinja2.ext.with_
silent=False`

**Comment by guoweikuang** [comment]
After upgrading setuptools (pip install --upgrade setuptools) from 20.7.0 to 35.0.1 it works now.

**Comment by akx** [comment]
Closing this as it seems upgrading `setuptools` will fix this issue. :)

**Comment by horejsek** [comment]
It happens again with setuptools 39.2.0, works until 39.1.0.

**Comment by JimmyTheNerd** [comment]
> It happens again with setuptools 39.2.0, works until 39.1.0.

Works with **setuptools 40.5.0**.

**Comment by trideceth12** [comment]
> > It happens again with setuptools 39.2.0, works until 39.1.0.
> 
> Works with **setuptools 40.5.0**.

Not for me.. I had to `pip install setuptools==39.1.0` to fix this ( I was on 40.5.0 before)
